### PR TITLE
12 order mgmt imp

### PR DESCRIPTION
**Pull request by ivantodorovich**
Hey @PierrickBrun 

I'm did some refactoring here to make the code more extensible.
I think it's even possible now to split the modules, if you still want to.

My goal here is to develop another module to manage 'draft' pos.orders as quotations, and it's going to depends on this one. This is why I needed some of these changes.

Please look if you have time

### CODE REVIEW

File: pos_order_mgmt/README.rst [hunk 1]
```diff
@@ -107,6 +107,9 @@ Contributors
 * David Vidal <david.vidal@tecnativa.com>
 * Sylvain LE GAL (https://twitter.com/legalsylvain)
 * Pierrick Brun <pierrick.brun@akretion.com>
+* Pierrick Brun <pierrick.brun@akretion.com>
```
Comment (PierrickBrun) [inline pos_order_mgmt/README.rst:110]: No need to credit me twice :smile:

**Comment by PierrickBrun** [comment]
That looks good, can you squash your commits before I merge it ?

File: pos_order_mgmt/README.rst (hunk 1 above)
Comment (ivantodorovich) [reply pos_order_mgmt/README.rst:110]: lol

**Comment by ivantodorovich** [comment]
done! 👍

**Pull request merged**